x-pack/plugins/monitoring_collection/server/routes/dynamic_route.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { Observable } from 'rxjs';
+import { schema } from '@kbn/config-schema';
+import { IRouter, ServiceStatus } from '../../../../../src/core/server';
+import { getESClusterUuid, getKibanaStats } from '../lib';
+import { MetricResult } from '../plugin';
+
+export function registerDynamicRoute({
+  router,
+  config,
+  overallStatus$,
+  getMetrics,
+}: {
+  router: IRouter;
+  config: {
+    allowAnonymous: boolean;
+    kibanaIndex: string;
+    kibanaVersion: string;
+    uuid: string;
+    server: {
+      name: string;
+      hostname: string;
+      port: number;
+    };
+  };
+  overallStatus$: Observable<ServiceStatus>;
+  getMetrics: (type: string) => Promise<MetricResult[] | MetricResult | undefined>;
+}) {
+  router.get(
+    {
+      path: `/api/monitoring_collection/{type}`,
+      options: {
+        // authRequired: !config.allowAnonymous,
```

Comment:
I'm not sure how we want to think about security here. Thoughts welcome!

Reply:
@jportner @legrego We're adding a new HTTP endpoint that will expose _somewhat_ arbitrary metrics about various systems within Kibana. Currently, we're only slated to only add metrics for the alerting framework ([exact details here](https://github.com/elastic/kibana/issues/122362#issuecomment-1020359743)) and none of these metrics expose any user-specific data (like named things). The intended consumer of this API will be Metricbeat, as part of the Stack Monitoring Metricbeat monitoring story. The other endpoint Metricbeat talks to is the `/api/stats` which exists in [the usage collection](https://github.com/elastic/kibana/blob/main/src/plugins/usage_collection/server/routes/stats/stats.ts#L73) plugin. Should we copy the same security settings for this new route? Should we be concerned about RBAC now or in the future for this route?